modules/apps/style-book/style-book-web-test/src/testIntegration/java/com/liferay/style/book/web/internal/portlet/action/test/UpdateStyleBookEntryPreviewMVCActionCommandTest.java

```diff
@@ -92,6 +93,60 @@ public void setUp() throws Exception {
 		_themeDisplay.setUser(TestPropsValues.getUser());
 	}
 
+	@Test(expected = NoSuchFileEntryException.class)
+	public void testInvalidStyleBookEntryPreview() throws Exception {
+		MockLiferayPortletActionRequest mockLiferayPortletActionRequest =
+			new MockLiferayPortletActionRequest();
+
+		StyleBookEntry styleBookEntry =
+			_styleBookEntryLocalService.addStyleBookEntry(
+				null, TestPropsValues.getUserId(), _group.getGroupId(), false,
+				StringPool.BLANK, RandomTestUtil.randomString(),
+				StringPool.BLANK, RandomTestUtil.randomString(),
+				_serviceContext);
+
+		Class<?> clazz = getClass();
+
+		FileEntry fileEntry = PortletFileRepositoryUtil.addPortletFileEntry(
+			null, _group.getGroupId(), TestPropsValues.getUserId(),
+			StyleBookEntry.class.getName(),
+			styleBookEntry.getStyleBookEntryId(), RandomTestUtil.randomString(),
+			_repository.getDlFolderId(),
+			clazz.getResourceAsStream(
+				"dependencies/frontend-tokens-values.json"),
+			"frontend-tokens-values.json", ContentTypes.APPLICATION_JSON,
+			false);
+
+		mockLiferayPortletActionRequest.addParameter(
+			"fileEntryId", String.valueOf(fileEntry.getFileEntryId()));
+
+		mockLiferayPortletActionRequest.addParameter(
+			"styleBookEntryId",
+			String.valueOf(styleBookEntry.getStyleBookEntryId()));
+		mockLiferayPortletActionRequest.setAttribute(
+			WebKeys.PORTLET_ID, StyleBookPortletKeys.STYLE_BOOK);
+		mockLiferayPortletActionRequest.setAttribute(
+			WebKeys.THEME_DISPLAY, _themeDisplay);
+
+		_updateStyleBookEntryPreviewMVCActionCommandTest.processAction(
+			mockLiferayPortletActionRequest, new MockActionResponse());
+
+		styleBookEntry = _styleBookEntryLocalService.fetchStyleBookEntry(
+			styleBookEntry.getStyleBookEntryId());
+
+		Assert.assertEquals(0, styleBookEntry.getPreviewFileEntryId());
+
+		Assert.assertTrue(
+			SessionErrors.contains(
+				mockLiferayPortletActionRequest, "invalidThumbnailFormat"));
+
+		PortletFileRepositoryUtil.getPortletFileEntry(
+			_group.getGroupId(), _repository.getDlFolderId(),
+			_getFileName(
+				fileEntry.getExtension(),
+				styleBookEntry.getStyleBookEntryId()));
```

Comment:
Usually assert instructions are the last part of the test. Is there a usage of this one? It seems it returns a file entry but no one use it

Reply:
@algobob  This call will throw `NoSuchFileEntryException`, it is stated at line 96.